compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEFusePackIntoLoop.cpp

```diff
@@ -122,29 +135,32 @@ void AMDAIEFusePackIntoLoopPass::runOnOperation() {
       return;
     }
 
-    SmallVector<tensor::ExtractSliceOp> sliceOps;
-    for (auto [index, operand] : llvm::enumerate(genericOp.getOperands())) {
-      FailureOr<tensor::ExtractSliceOp> sliceOp =
+    // Materialize each slice of the producer in place.
+    for (Value operand : genericOp.getOperands()) {
+      FailureOr<tensor::ExtractSliceOp> maybeSliceOp =
           getTensorExtractSliceDefiningOp(operand);
-      if (!failed(sliceOp)) {
-        sliceOps.push_back(sliceOp.value());
+      if (succeeded(maybeSliceOp)) {
+        tensor::ExtractSliceOp sliceOp = maybeSliceOp.value();
+        std::optional<scf::SCFFuseProducerOfSliceResult> fusedProducer =
+            scf::tileAndFuseProducerOfSlice(rewriter, sliceOp,
+                                            MutableArrayRef(&loops, 1));
+        if (!fusedProducer) {
+          funcOp->emitOpError("Failed to fuse pack ops into for loop.");
+          return signalPassFailure();
+        }
       }
-    }
 
-    if (sliceOps.empty()) {
-      LLVM_DEBUG(llvm::dbgs() << "----- Pack ops are already fused or no slice "
-                                 "ops were found.-----\n");
-      return;
-    }
-
-    // Materialize each slice of the producer in place.
-    for (auto sliceOp : sliceOps) {
-      std::optional<scf::SCFFuseProducerOfSliceResult> fusedProducer =
-          scf::tileAndFuseProducerOfSlice(rewriter, sliceOp,
-                                          MutableArrayRef(&loops, 1));
-      if (!fusedProducer) {
-        funcOp->emitOpError("Failed to fuse pack ops into for loop.");
-        return signalPassFailure();
+      // Case where operand of generic is a pack op which is in a different
+      // block than the generic's block.
+      else {
+        if (auto parent =
+                dyn_cast_if_present<tensor::PackOp>(operand.getDefiningOp())) {
+          Block *genericBlock = genericOp->getBlock();
```

Comment:
I think here you need to use `else if`, and then add another `else` in which adding some debug message and return.

Reply:
Logically the else condition is not needed, but I was thinking it may good to add some debug message so people know why the pack is not fused into the loop in some specific situation. You can decide whether to add it.

Reply:
The operand might not even come from a pack though, it could be anything...